pdr_backend/lake/etl.py

```diff
@@ -56,27 +77,62 @@ def _drop_temp_sql_tables(self):
         db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
         for dataclass in _ETL_REGISTERED_LAKE_TABLES:
             table_name = dataclass.get_lake_table_name()
-            db.drop_view(ETLTable(table_name).fullname)
-            db.drop_table(TempTable(table_name).fullname)
+            db.drop_table(TempTable(table_name).table_name)
+            db.drop_table(NewEventsTable(table_name).table_name)
+            db.drop_table(UpdateEventsTable(table_name).table_name)
+            db.drop_table(TempUpdateTable(table_name).table_name)
 
-    def _move_from_temp_tables_to_live(self):
+    def _do_bronze_swap_to_prod_atomic(self):
         """
         @description
-            Move the records from our bronze temporary tables to live, in-production tables
+            Merge the bronze ETL tables against prod database
+            This needs to become a single transaction per table
+            Such that it remains atomic, and is able to resume if it fails
         """
-
         db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
-        for dataclass in _ETL_REGISTERED_LAKE_TABLES:
-            table_name = dataclass.get_lake_table_name()
-            logger.info("move table %s to live", table_name)
-            temp_table = TempTable(table_name)
-            permanent_table = NamedTable(table_name)
 
-            if db.table_exists(temp_table.fullname):
-                db.move_table_data(temp_table, permanent_table)
+        bronze_tables = [BronzePrediction]
+        for table in bronze_tables:
+            prod_table = Table.from_dataclass(table)
+            new_events_table = NewEventsTable.from_dataclass(table)
+            update_events_table = UpdateEventsTable.from_dataclass(table)
+            temp_update_table = TempUpdateTable.from_dataclass(table)
+
+            if db.table_exists(update_events_table.table_name):
```

Comment:
I think you can do an "if inversion" here. If the table doesn't exist, `continue`. The subsequent block can then be unindented.

Reply:
done, pushed to => https://github.com/oceanprotocol/pdr-backend/pull/1423